crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -132,6 +153,15 @@ impl ProtocolFees {
             .collect_vec();
         boundary::order::to_domain(order, protocol_fees)
     }
+
+    /// Internal function in `ProtocolFees` in order to determine whether the
+    /// driver has to apply the protocol fee to an order/solution
+    fn owner_exempt_from_fees(&self, address: H160) -> bool {
+        // If the address is whitelisted -> the owner is excempt from fees
+        self.protocol_fee_discounted_addresses
+            .get(&address)
+            .is_some()
```

Comment:
``` suggestion
            .contains(&address)
```

Reply:
Not sure if it's worth a separate function.

Reply:
Yeah, especially with your suggestion to use `.contains()` inlining this logic still leads to very readable code.